### PR TITLE
Raise client timeout to 30s

### PR DESCRIPTION
**Pull request by patrislav**

### CODE REVIEW

File: cmd/waas-auth/main.go [hunk 1]
```diff
@@ -40,17 +40,14 @@ func main() {
...
-		Timeout: 10 * time.Second,
-		Transport: transport.Chain(
-			baseTransport,
-			transport.SetHeader("User-Agent", "waas-authenticator/"+waasauthenticator.VERSION),
-			traceid.Transport,
-		),
-	}
+	// HTTP transport chain to use for all outgoing connections out of the enclave
+	transportChain := transport.Chain(
+		baseTransport,
```
Comment (c2h5oh) [inline cmd/waas-auth/main.go:45]: just to confirm: baseTransport has 30s timeout?

Comment (patrislav) [reply cmd/waas-auth/main.go:45]: baseTransport is defined above, the only timeouts it specifies are the idle and TLS handshake. The actual request timeout is defined on the client in `rpc.New`: https://github.com/0xsequence/waas-authenticator/pull/69/files/cd191a960933e9dacc9365af4777e1d85c0671ac#diff-6c518b6e5342273a8ecca52a51a6a7398d6fa4596f53f425212797def8819564R80

**Review by c2h5oh — APPROVED** [review]

File: rpc/rpc.go [hunk 2]
```diff
@@ -75,9 +75,10 @@ type RPC struct {
...
 }
 
-func New(cfg *config.Config, client *http.Client) (*RPC, error) {
-	if client == nil {
-		client = &http.Client{Timeout: 10 * time.Second}
+func New(cfg *config.Config, transport http.RoundTripper) (*RPC, error) {
+	client := &http.Client{
+		Timeout:   30 * time.Second,
+		Transport: transport,
 	}
```
Comment (VojtechVitek) [inline rpc/rpc.go:82]: Imho it'd be better if you passed `*http.Client` as before and defined the timeout in some parent function?

Or what is the reason you've switched to passing transport instead?

**Review by VojtechVitek — COMMENTED** [review]
Minor comment. LGTM otherwise

File: rpc/rpc.go (hunk 2 above)
Comment (patrislav) [reply rpc/rpc.go:82]: I wanted to have the Timeout configured in a single place as it's cleaner this way. I also thought I'd be able to add a test for it (and I still would like that) but in the current setup we use a mock in place of Stack API. I'd like to move to running a httptest server instead to use the client, this way we'd be able to better test it. But this work will have to wait

**Review by VojtechVitek — APPROVED** [review]

**Pull request merged**